packages/router/src/grpc/offscreen-client.ts

```diff
@@ -15,59 +15,70 @@ const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
 let active = 0;
 
 const activateOffscreen = async () => {
-  const offscreenExists = await chrome.runtime
+  const noOffscreen = chrome.runtime
     .getContexts({
       contextTypes: [chrome.runtime.ContextType.OFFSCREEN_DOCUMENT],
     })
-    .then(contexts => contexts.length > 0);
+    .then(offscreenContexts => !offscreenContexts.length);
 
-  if (!active || !offscreenExists) {
-    await chrome.offscreen.createDocument({
-      url: chrome.runtime.getURL(OFFSCREEN_DOCUMENT_PATH),
-      reasons: [chrome.offscreen.Reason.WORKERS],
-      justification: 'Manages Penumbra transaction WASM workers',
-    });
-  }
-
-  active++;
+  if (!active++ || (await noOffscreen))
```

Comment:
Could we move the `await` back up above?

```suggestion
  if (!active++ || noOffscreen)
```

I'd also consider renaming `noOffscreen` to be a bit more descriptive so that we can understand what it means at a glance — e.g., `offscreenWorkerDoesNotExist` or something

Reply:
Hm, I'd also consider incrementing `active` _inside_ the `if` block. This line is packing a lot into it, which makes it a little harder to grok. It's:
- Checking that there are 0 active offscreen contexts
- Incrementing the count of offscreen contexts
- `await`ing `noOffscreen` and then checking its value

May just be my preference, but I feel like an `if` statement is generally clearest if it's just evaluating existing values, rather than assigning new ones.

Reply:
we discussed this while pairing on discord and i agree this does need more explanation but it's this way for a reason. would like to get off the rebase treadmill so i'll issue this and handle it later